### DISCUSSION TITLE
QZ Companion App for Garmin

### DISCUSSION THREAD

**Discussion by nieko-nera** [cagnulein/qdomyos-zwift#1395] [Q&A] [answered]
Hello, how can I use this [QZ Companion App](https://apps.garmin.com/en-US/apps/3746ce54-a9e9-42b0-9be9-b867f8d20f7d?fbclid=IwAR3tGY_fAWzFHyGGzBmogInGCk5gbiRiH0iDiCRwT38n3TfV7NfdZHqCsOk) app? I installed this app, opened, it's shows my HR on watch screen.  What settings I need to enable in QZ app on smartphone to see my HR? My watch is vivoactive 3

**Comment by cagnulein** [comment]
@nieko-nera I didn't merge yet #1330 I will try to do this in the next days and then you will see a new setting for the companion in the QZ app

**Comment by nieko-nera** [comment]
QZ version 2.13.26 on Android not working for me. I run QZ companion app on watch, Garmin watch companion options enabled on QZ android.
When I open companion app on watch, should I see something with nRF Connect?

**Comment by cagnulein** [comment]
no @nieko-nera the companion talks with the qz by bluetooth on an hidden channel. Just be sure:
- garmin watch companion setting enabled
- heart rate belt name setting set to disabled
- version 1.2.1 of the companion

I'm adding also cadence to the companion so I will need to release another QZ version to handle it.
Let me know

**Reply by nieko-nera** [reply]
> * garmin watch companion setting enabled
> * heart rate belt name setting set to disabled

yes
> * version 1.2.1 of the companion

version 1.2.0, no update available
[debug-Thu_Apr_6_22_05_46_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/11173790/debug-Thu_Apr_6_22_05_46_2023.log)

maybe I need to try iOS version?

**Reply by cagnulein** [reply]
do you see hr on the garmin watch? do you have garmin connect app on the same android connected to the watch?

**Reply by nieko-nera** [reply]
i see hr on watch, garmin connect is on iPhone, QZ is on android

**Reply by cagnulein** [reply, marked as answer]
@nieko-nera qz and garmin connect must be on the same device, and also garmin connect must be connected to the watch. if you need the beta for ios send me an email

**Reply by nieko-nera** [reply]
Its working! Thank You! 🥇

**Reply by cagnulein** [reply]
great! I'm uploading 2.13.27 with companion 1.2.1 that support also running cadence!

**Reply by ryegye24** [reply]
> @nieko-nera qz and garmin connect must be on the same device, and also garmin connect must be connected to the watch. if you need the beta for ios send me an email

Would it be possible to get this to work with GadgetBridge rather than the Garmin Connect app on the device running QZ? That would be a game changer for me

**Comment by chreds** [comment]
For anyone else having trouble, I had to turn off broadcast heart rate during activity on the garmin watch. Maybe it doesn't like doing ant+ hr broadcast at the same time as the companion app.

**Reply by dstrimble** [reply]
I'm gonna give this a try.

**Reply by dstrimble** [reply]
Still nothing. So if qz companion is connected I should see heart rate in qz?

My setup was, qz and connect running on same phone. Qz companion on my Garmin app. Heart rate disabled in QZ and broadcast disabled on my watch. Any more suggestions?

**Reply by cagnulein** [reply]
share a debug log @dstrimble

**Reply by dstrimble** [reply]
How do I get that?

**Reply by cagnulein** [reply]
https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F

**Reply by dstrimble** [reply]
[debug-Wed_Dec_13_06_52_38_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/13661453/debug-Wed_Dec_13_06_52_38_2023.log)

Here it is

**Reply by cagnulein** [reply]
hi @dstrimble in this log at least the ant_heart setting is wrong, you need to disable it and restart qz. Let me know and in case provide me another debug log. thanks

**Reply by dstrimble** [reply]
That seemed to work better, however, I assumed it would auto save the activity to my Garmin connect account and it didn't. Is there anything special I have to do when I'm closing the ride? Instead I uploaded the fit file, but in that case I don't know think it effects your training load on Garmin.

**Reply by cagnulein** [reply]
@dstrimble the companion is only to send HR to your device. If you need to save the activity on the garmin to affect the training load, i suggest to use ANT+ from QZ to Garmin since it's the only way how Garmin accepts metrics from QZ or any other 3rd party apps. more info here https://robertoviola.cloud/2023/07/06/connect-any-indoor-bike-with-your-garmin-device/

**Reply by dstrimble** [reply]
I've got an ant + dongle on order now, thank you for the information. Do I have to do anything to complete the ride and have it automatically load to Garmin Connect, once I do have that dongle?

**Reply by cagnulein** [reply]
answer here https://github.com/cagnulein/qdomyos-zwift/discussions/1395#discussioncomment-7865285 :)
anyway when you have the dongle you have
- downlaod ant+ checker tools from the play store and install all the driver
- enable ant cadence in qz
- pair on your garmin device, the qz app via ant+ (you will see the qz sensors)
- then just start a garmin workout as you usually do outside

Let me know!

**Reply by dstrimble** [reply]
ant+ checker isn't compatible with my phone, is there another option?

**Reply by chreds** [reply]
ANT+ checker wasn't compatible with mine either but I just installed the other ant+ apps and it works (ANT+ USB, ANT+ Plugin manager, and there as another one I think. It's all under the same app developer "ANT+").

**Reply by cagnulein** [reply]
yes download these app so
let me know

**Reply by dstrimble** [reply]
Can you share an amazon link to the ant dongle you show in that article?

**Reply by cagnulein** [reply]
@dstrimble any ant+ usb dongle
works

**Reply by dstrimble** [reply]
Ok. I was reading more of the details on what I ordered and it didn't seem like it would work on a phone, but I'll give it a go.

**Reply by cagnulein** [reply]
@dstrimble you need also a cable adaptor to plug the classic usb into a micro usb on your phone or a usb type c

**Reply by dstrimble** [reply]
Yep. I have a few of those laying around already.

**Reply by dstrimble** [reply]
@cagnulein this worked perfectly with the ant + dongle. All my data streamed to my Garmin Enduro watch and Garmin Connect app.

![image](https://github.com/cagnulein/qdomyos-zwift/assets/10944604/eef7fe53-3a7c-425a-a02a-7e17dbf24936)

This is the dongle I used: https://a.co/d/iXAW9iL

**Reply by cagnulein** [reply]
that's amazing @dstrimble

**Reply by dstrimble** [reply]
Actually, might be a slight problem. I definitely didn't go 80 mph. The distance is way over calculating. Any thoughts?

**Comment by cagnulein** [comment]
Thanks for the tip!

Il giorno lun 25 set 2023 alle 18:54 chreds ***@***.***> ha
scritto:

**Comment by chreds** [comment]
That'll get setup in your ride app. In Rouvy you can connect your Garmin
Connect account on the Rouvy website and have set it up to automatic
upload. Not sure if the other ones support auto upload to Garmin.

**Reply by dstrimble** [reply]
I don't like the third party ride apps (cost, for one). According to what Roberto said I should be able to just use the indoor biking activity on my garmin once QZ recognized as a sensor

**Reply by chreds** [reply]
I had limited success going back the other way but you might be able to. There were some settings in QZ to output cadence and power over ANT, but I couldn't get that working reliably with an Android device in between. Maybe Apple works better.

Having the ride app submit the ride has been much more reliable for me. I've really enjoyed Rouvy and it's a lot cheaper than Peloton. They had a decent black friday deal for the year. Maybe they will have a boxing day one too.

**Reply by chreds** [reply]
It looks like maybe the companion app is receiving some updates though since I last used it. If it's passing cadence, etc. to connect as well as heartbeat then that could be a good solution.

**Reply by cagnulein** [reply]
@chreds the companion app can send only data to qz not to the garmin connect. is what you meant?

**Reply by chreds** [reply]
"I'm adding also cadence to the companion so I will need to release another QZ version to handle it.
Let me know"

I thought maybe this meant:
1. QZ receives cadence from bike
2. QZ sends cadence to garmin watch over companion app
3. Companion app simulates activity on watch (i've seen a few other apps that do this like myswimpro.com, or https://apps.garmin.com/en-US/apps/3ae9729f-c364-451b-9ca4-ca540749f6e7)

But I think #2 is probably not possible to simulate a sensor that's not connected natively on the watch? I think those apps maybe create a their own FIT file of the activity or something.

Alternatively:

This should be possible if you want to control the activity from the watch:
1. QZ receives cadence, speed from bike
2. QZ sends cadence, speed to watch over ANT+
3. Use normal indoor biking activity on garmin watch

Don't even need the companion app there. I think companion app is for sending to QZ if you don't have ANT+, right? And the cadence you refer to above is the running cadence for a running activity?

**Reply by cagnulein** [reply]
> "I'm adding also cadence to the companion so I will need to release another QZ version to handle it. Let me know"
> 
> I thought maybe this meant:
> 
> 1. QZ receives cadence from bike

nope running cadence I was meaning :)

> 2. QZ sends cadence to garmin watch over companion app
> 3. Companion app simulates activity on watch (i've seen a few other apps that do this like myswimpro.com, or https://apps.garmin.com/en-US/apps/3ae9729f-c364-451b-9ca4-ca540749f6e7)

yes qz already simulate an activity but Garmin doesn't allow 3rd party apps to save custom metrics, i mean metrics that are not recording DIRECTLY from the garmin itself

> 
> But I think #2 is probably not possible to simulate a sensor that's not connected natively on the watch? I think those apps maybe create a their own FIT file of the activity or something.

exactly

> 
> Alternatively:
> 
> This should be possible if you want to control the activity from the watch:
> 
> 1. QZ receives cadence, speed from bike
> 2. QZ sends cadence, speed to watch over ANT+
> 3. Use normal indoor biking activity on garmin watch

exactly, it's what i was suggesting to @dstrimble 

> 
> Don't even need the companion app there. I think companion app is for sending to QZ if you don't have ANT+, right? And the cadence you refer to above is the running cadence for a running activity?

exactly :)

**Comment by cagnulein** [comment]
yes you have to set the wheel circumference to 100mm

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 22 dic 2023 alle 16:33 Daniel Trimble <
***@***.***> ha scritto:

**Reply by dstrimble** [reply]
Can you point me to this option?

**Reply by cagnulein** [reply]
it's on your garmin device, in the settings of the qz sensor

**Reply by OrislavK** [reply]
where are this option? can't find it. Settings of sensor QZ doesn't have such option.

**Reply by cagnulein** [reply]
some garmin watches doesn't have this, usually when they are only for runners. I have it on my garmin edge for example. Check in the manual of your watch @OrislavK

**Reply by OrislavK** [reply]
There are diffirents exist in profile of just bike and bike indoor profile, i can't find setting of wheel size in starting bike indoor profile and in settings QZ sensor. In QZ sensor only status, calibration  prompt, calibrate,  about, name, remove.

**Reply by cagnulein** [reply]
@OrislavK i'm talking about this 
![image](https://github.com/cagnulein/qdomyos-zwift/assets/48216/5f3c49bb-8c6f-4318-8c29-a532371344c9)

on my edge this is available when i'm sending ANT+ metrics from QZ on my android to the garmin. The diffence on your setup is that you are using bluetooth.

**Comment by OrislavK** [comment]
Can QZ get HR from watches and retranslate it to Zwift? How? Zwift found Wahoo HR, but there are zero.

**Reply by cagnulein** [reply]
@OrislavK as I said you have to use the qz garmin companion app and enable the garmin companion setting in QZ as well. garmin connect must be installed and paired on the device where you would like to collect the HR (for example iphone)

**Reply by OrislavK** [reply]
I have garmin fenix 6 pro.
I have iPhone
I install  qz garmin companion app on watches. qz garmin companion app doesn't have any settings nor via garmin connect. nor via watches.

![C0476C32-FE06-4FAF-B1E7-B1557F87249B_4_5005_c](https://github.com/cagnulein/qdomyos-zwift/assets/25452579/c3c20c45-23cd-4ff4-af6f-1bfb439fba9d)

I install on iphone test version of qdomyos with garmin support  (version 2.16.30). In Garmin options Garmin Sensor and Companion App was enabled both. (also this is done on Mac). Settings saved and APP restarted.
I open QZ app on watches and it show 3 rows - HR, STEP and timer string. HR show actual HR.
While QZ app on watches opened, i press left centered button and see 2 rows with names item1 and item2. Switching between them does nothing.

In qdomyos HR show 0.

**Reply by cagnulein** [reply]
> I install qz garmin companion app on watches. qz garmin companion app doesn't have any settings nor via garmin connect. nor via watches.

yes that's right, the setting is on the iphone. just press the hw start button on the watch in the companion app. when you press the start button you have to see the HR on the watch while the elapsed timer is counting.

In the qz on iphone, only the Companion App setting is needed. the other one is to send via bluetooth metrics on the garmin from the mac

**Reply by OrislavK** [reply]
> yes that's right, the setting is on the iphone. just press the hw start button on the watch in the companion app. when you press the start button you have to see the HR on the watch while the elapsed timer is counting.
> 

yes, then open companion app i see current HR and then press again hw button start - counter started, meanwhile on iphone HR are 0.0, as and on Mac.
Also, i planned start indoor bike training on garmin watches with Yesoul G1MAX and in same time i want see HR in ZWIFT from watches/Qdomyos which must be run on Mac for now and for test i run QD on iphone and it opened, but no HR from watches on iphone/mac and zwift. Is it possible to get HR in Zwift? How? :) this is just not working.

**Reply by cagnulein** [reply]
this can work only on iphone for sure, not macos, since you have install and pair garmin connect on the watch. So provide a debug log from the iphone while you're doing this so i can check the issue that you have.

The right setup to send hr from your garmin to zwift is the iphone qz version.

**Reply by OrislavK** [reply]
> So provide a debug log from the iphone while you're doing this so i can check the issue that you have.
> 

where are to get debug log?

**Reply by cagnulein** [reply]
https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F

**Comment by cagnulein** [comment]
Ask them to integrate ftms fitness device. It's a standard!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 28 ott 2024 alle 18:47 ryegye24 ***@***.***> ha
scritto:
